src/meta/src/manager/env.rs

```diff
@@ -55,7 +56,7 @@ where
     /// idle status manager.
     idle_manager: IdleManagerRef,
 
-    info: MetaLeaderInfo,
+    leader_rx: WatchReceiver<(MetaLeaderInfo, bool)>,
```

Comment:
Can we just use a RwMutex for the leader infos? So we can simply read and update it with some internal functions.

Reply:
I disagree with this proposal. IMHO this is already a really simple version, since we never actually access the `Receiver` from the outside. To get the current leader, we simply use `get_leader_info(&self) -> MetaLeaderInfo`. 

Would using a mutex and some centrally managed state not complicate this unnecessarily?

Reply:
```rust
    pub struct MetaSrvEnv {
        ...
        info: Arc<RwLock<MetaLeaderInfo>>,
    }

    impl MetaSrvEnv {
        pub async fn get_leader_info(&self) -> MetaLeaderInfo {
            self.info.read().await.clone()
        }
     
        pub async fn update_leader_info(&self, info: MetaLeaderInfo) {
            let mut guard = self.info.write().await;
            *guard = info;
        }
    }
```
This looks more intuitive and simple than using a channel.

Reply:
As I commented in `start_leader_srv` part, updating leader info seems no longer needed.

Reply:
So far I kept the version where we do not update the leader info. I would agree that updating the leader info is no longer needed.